modules.d/95znet/parse-ccw.sh

```diff
@@ -4,4 +4,24 @@ for ccw_arg in $(getargs rd.ccw -d 'rd_CCW=') $(getargs rd.znet -d 'rd_ZNET=');
     echo $ccw_arg >> /etc/ccw.conf
 done
 
+for ifname in $(getargs rd.znet_ifname); do
+    echo "$ifname" | {
+        IFS=: read ifname_if ifname_subchannels _
+        if ! [ -n "$ifname_if" ] && [ -n "$ifname_subchannels" ] && [ -z "$_" ]; then
```

Comment:
I don't think the check is correct. 
[lnykryn@X1 dracut(znet)]$ ifname_if=aaa
[lnykryn@X1 dracut(znet)]$ ifname_subchannels=""
[lnykryn@X1 dracut(znet)]$ _=""
[lnykryn@X1 dracut(znet)]$ if ! [ -n "$ifname_if" ] && [ -n "$ifname_subchannels" ] && [ -z "$_" ]; then echo fail; fi
[lnykryn@X1 dracut(znet)]$ 


I would use 
if [ -z "$ifname_if" ] || [ -z "$ifname_subchannels" ]  || [ -n "$_" ]

Reply:
Right, I messed up with the operator priority, thanks for pointint it out.